Running eqy can final report error "Argument list too long"

Problem 1:

For the two provided designs the runtimes are as follows:

```
Design  Instance count    Runtime (simple)   Runtime (sby)
aes         32k                37m             257m
jpeg        55k               140m             900m 

```
It is not clear if these run times are expected or is there something incorrect with the way we are using eqy.

Problem 2:

At the end of all the four runs (2 designs/2 different strategies) an error is encountered before creating the final report. The logfile for one run is available in aes/logfile.txt. Here is the last few lines of the logfile:

```
EQY 15:57:04 [aes] run: Running strategy 'sby' on 'aes_cipher_top.wire389.Y'..
EQY 15:57:05 [aes] run: Proved equivalence of partition 'aes_cipher_top.wire389.Y' using strategy 'sby'
EQY 15:57:05 [aes] run: make -f strategies.mk summary
EQY 15:57:05 [aes] run: make[1]: Entering directory '/home/harsh/bugs/eqy/aes'
EQY 15:57:05 [aes] run: make[1]: execvp: /bin/sh: Argument list too long
EQY 15:57:05 [aes] run: make[1]: *** [strategies.mk:825391: summary] Error 127
EQY 15:57:05 [aes] run: make[1]: Leaving directory '/home/harsh/bugs/eqy/aes'
EQY 15:57:05 [aes] run: make: *** [strategies.mk:825389: all] Error 2
EQY 15:57:05 [aes] run: make: Leaving directory '/home/harsh/bugs/eqy/aes'
EQY 15:57:05 [aes] run: finished (returncode=2)
```


Testcase data is public. The designs are [here](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/tree/master/flow/designs/asap7). And the library data is [here](https://github.com/The-OpenROAD-Project/asap7)

--- a/src/eqy.py
+++ b/src/eqy.py
@@ -1080,19 +1080,23 @@ def make_scripts(args, cfg, job, strategies):
                 exit_with_error(f"No configured strategy supports partition {partition.name}")
             final_targets.append(prev_strategy)
 
+        with open(f"{args.workdir}/summary_targets.list", "w") as targets_f:
+            for target in final_targets:
+                print(target, file=targets_f)
+
         prmkf(f".PHONY: all summary")
         prmkf(f"all: {' '.join(sorted(final_targets))}")
         prmkf(f"\t$(MAKE) -f strategies.mk summary")
         prmkf(f"""summary:""")
         prmkf(f"""\t@rc=0 ; \\""")
-        prmkf(f"""\tfor f in {" ".join(final_targets)} ; do \\""")
+        prmkf(f"""\twhile read f; do \\""")
         prmkf(f"""\t\tp=$${{f#strategies/}} ; p=$${{p%/*/status}} ; \\""")
         prmkf(f"""\t\tif grep -q "PASS" $$f ; then \\""")
         prmkf(f"""\t\t\techo "* Successfully proved equivalence of partition $$p" ; \\""")
         prmkf(f"""\t\telse \\""")
         prmkf(f"""\t\t\techo "* Failed to prove equivalence of partition $$p" ; rc=1 ; \\""")
         prmkf(f"""\t\tfi ; \\""")
-        prmkf(f"""\tdone ; \\""")
+        prmkf(f"""\tdone < summary_targets.list ; \\""")
         prmkf(f"""\tif [ "$$rc" -eq 0 ] ; then \\""")
         prmkf(f"""\t\techo "* Successfully proved designs equivalent" ; \\""")
         prmkf(f"""\tfi""")

